core-bundle/tests/PhpunitExtension/GlobalStateWatcher.php

```diff
@@ -0,0 +1,92 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of Contao.
+ *
+ * (c) Leo Feyer
+ *
+ * @license LGPL-3.0-or-later
+ */
+
+namespace Contao\CoreBundle\Tests\PhpunitExtension;
+
+use PHPUnit\Framework\ExpectationFailedException;
+use PHPUnit\Runner\AfterTestHook;
+use PHPUnit\Runner\BeforeTestHook;
+use SebastianBergmann\Diff\Differ;
+use SebastianBergmann\Diff\Output\UnifiedDiffOutputBuilder;
+
+final class GlobalStateWatcher implements AfterTestHook, BeforeTestHook
```

Comment:
With this extension you can execute

```sh
vendor/bin/phpunit --extensions=Contao\\CoreBundle\\Tests\\PhpunitExtension\\GlobalStateWatcher
```

and you will get an error if a test leaks global state via `$GLOBALS` or static class properties.